### DISCUSSION TITLE
Import block not showing correct name in some Docs pages

### DISCUSSION THREAD

**Discussion by msanzv** [storybookjs/storybook#25255] [Help]
### Describe the bug

We have a design system documentation with over 50 components, and in all of them, we use the `<Import>` block to start our Docs pages like this:

<img width="1062" alt="image" src="https://github.com/storybookjs/storybook/assets/1478898/4090c24c-d89f-4e93-a0e7-2c4d78683757">

But in a few components, it renders like this instead:

<img width="1041" alt="image" src="https://github.com/storybookjs/storybook/assets/1478898/68d25101-ee01-4042-b1e3-ced8022bfe8b">
<img width="1046" alt="image" src="https://github.com/storybookjs/storybook/assets/1478898/c32a427b-fa14-4e53-b5b3-5512a4ac433b">

Some additional info we already checked:
- It only happens in 3 components: 2 of them show `o` and the other shows `Q`
- The characters don't change from build to build, they're always those
- **It only happens in production & preview, not in local**
- The Import blocks or the stories they're in don't have anything (meaningful) different than the stories that work fine 
- It happened with storybook 6 (csf2), and keeps happening now that we moved them to storybook 7 (csf3)

With all of the above, my guess is that it doesn't have anything to do with the stories or the mdx page, so I turned my attention to the components themselves, to see if they could be formed or exported in a different way than others. And maybe it's that, but I can't see how they're different, apart from the obvious things. 

These are the three components (simplified for readability) that show this behaviour:

```tsx
export const SSRSupport = () => (
  <style dangerouslySetInnerHTML={{ __html: getCssText() }} />
);
```

```tsx
const FormControl = ({
  // more props
  ...rest
}: FromControlProps) => (
  <div className={cx(FormControlStyles.root({ invalid }), className)} {...rest}>
    {Children.map(children, (child) => {
      const nodes = removeFragment(child);
      return Children.map(nodes, (node) =>
        isValidElement(node)
          ? cloneElement(node,  { disabled, invalid, required } as any,)
          : node,
      );
    })}
  </div>
);

export { FormControl };
```

```tsx
const Root = styled(RadioGroupPrimitive.Root, RadioGroupStyles.Root);

const RadioGroup = ({
  ...props
}: RadioGroupProps) => {
  const [state, setState] = useControllableState({
    // initialization
  });

  return (
    <RadioScope value={state}>
      <Root {...props}  >
        {children}
      </Root>
    </RadioScope>
  );
};

export { RadioGroup };
```

We have other components with very similar structures that are not showing this behaviour, so I'm a bit lost. Any idea what could be happening here? I tried looking here in the issues and in google for similar issues but couldn't find anything, maybe because I couldn't figure out which keywords to use.

Any help would be appreciated, thanks in advance!

### To Reproduce

I can't even reproduce it myself, sorry. This is more a question than a bug report I guess.

### System

```bash
Storybook Environment Info:

  System:
    OS: macOS 14.2
    CPU: (10) arm64 Apple M1 Pro
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.18.2 - ~/.nvm/versions/node/v18.18.2/bin/node
    Yarn: 4.0.2 - ~/.nvm/versions/node/v18.18.2/bin/yarn <----- active
    npm: 10.2.1 - ~/Sites/only-ui/node_modules/.bin/npm
  Browsers:
    Chrome: 120.0.6099.62
    Safari: 17.2
  npmPackages:
    @storybook/addon-a11y: ^7.5.3 => 7.5.3 
    @storybook/addon-essentials: ^7.5.3 => 7.5.3 
    @storybook/addon-interactions: ^7.5.3 => 7.5.3 
    @storybook/addon-links: ^7.5.3 => 7.5.3 
    @storybook/blocks: ^7.5.3 => 7.5.3 
    @storybook/cli: ^7.5.3 => 7.5.3 
    @storybook/jest: ^0.2.3 => 0.2.3 
    @storybook/react: ^7.5.3 => 7.5.3 
    @storybook/react-vite: ^7.5.3 => 7.5.3 
    @storybook/test-runner: ^0.15.1 => 0.15.1 
    @storybook/testing-library: ^0.2.2 => 0.2.2 
    @storybook/theming: ^7.5.3 => 7.5.3 
    eslint-plugin-storybook: ^0.6.15 => 0.6.15 
    storybook: ^7.5.3 => 7.5.3
```

**Comment by shilman** [comment]
The `Import` block is not part of Storybook's core. Can you share the code for that?

Have you tried disabling minification in your viteConfig?

```js
// main.js
export default {
  // stories, addons, etc...
  async viteFinal(config) {
    return mergeConfig(config, {
      build: {
        minify: false,
      },
    });
  },
}
```

**Reply by msanzv** [reply]
> The Import block is not part of Storybook's core. Can you share the code for that?

Sorry, forgot to mention, our `Import` block is just a `Source` block underneath:

```tsx
import { Source } from '@storybook/addon-docs';
import { ComponentType } from 'react';
import packageJson from '../../../package.json';

type ImportProps = {
  component: ComponentType;
};

const Import = ({ component }: ImportProps) => {
  const name = component.displayName || component.name;

  return (
    <Source
      language="js"
      dark={true}
      code={`import { ${name} } from "${packageJson.name}"`}
    />
  );
};

export { Import };
```

> Have you tried disabling minification in your viteConfig?

I will give that a try tomorrow

**Reply by msanzv** [reply]
Hi @shilman , I was able to test the `minify: false` option and in the case of the 2 components showing `{ o }` now it shows correctly, while in the case of `RadioGroup` which was showing a `{ Q }` now it shows this:

<img width="1101" alt="image" src="https://github.com/storybookjs/storybook/assets/1478898/29cda9f3-59b2-4d74-b7c9-b3432d4509f7">

So the different letters did indeed mean that two different things are happening to them... Now I'm even more intrigued 🤔 

This is our full `main.ts`, in case it helps, though it's pretty simple:

<details>

<summary>Show main.ts</summary>

```ts
import { StorybookConfig } from '@storybook/react-vite';
import path from 'node:path';
import vite from 'vite';

const config: StorybookConfig = {
  stories: [
    '../docs/guides/**/*.stories.*',
    '../docs/guides/**/*.mdx',
    '../src/**/*.stories.*',
    '../src/**/*.mdx',
  ],

  addons: [
    '@storybook/addon-essentials',
    '@storybook/addon-interactions',
    '@storybook/addon-links',
    '@storybook/addon-a11y',
  ],

  framework: '@storybook/react-vite',

  async viteFinal(config) {
    // Merge custom configuration into the default config
    return vite.mergeConfig(config, {
      build: {
        minify: false,
      },
      esbuild: {
        supported: {
          'top-level-await': true,
        },
      },
      resolve: {
        alias: {
          '@/docs': path.resolve(__dirname, '../docs'),
          '@': path.resolve(__dirname, '../src'),
        },
      },
    });
  },

  docs: {
    autodocs: true,
  },
};
``` 

</details>

The only peculiarity that I found in `RadioGroup.tsx`, that I simplified in my original post but could be relevant, is that it gets typed and exported a bit different to all others, like this:

```ts
const Root = styled(RadioGroupPrimitive.Root, RadioGroupStyles.Root);

const RadioGroup = ({
  ...props
}: RadioGroupProps) => {
  const [state, setState] = useControllableState({
    // initialization
  });

  return (
    <RadioScope value={state}>
      <Root {...props}  >
        {children}
      </Root>
    </RadioScope>
  );
};

type CompoundedType = typeof RadioGroup & {
  Item: typeof RadioGroupItem;
  Card: typeof RadioCard;
};

const CompoundedRadioGroup = RadioGroup as CompoundedType;
CompoundedRadioGroup.Item = RadioGroupItem;
CompoundedRadioGroup.Card = RadioCard;

export { CompoundedRadioGroup as RadioGroup };
```

Not sure why it's the only one handled like that, it was done before I joined the team and we don't have that pattern in other components. Not sure if it could be relevant to this, but just in case.

The other 2 components (the ones showing an `{ o }` when minified) don't seem to have any peculiarity like this, just div wrappers to add styles, which we use extensively in other components.

**Reply by msanzv** [reply]
Just as addition info in case it's useful: I just noticed this behaviour is not limited to that `Import block`, it also happens in a couple other places. For example, in the 'Show Code' section of our `ComboBox` component, I see this in local, or with minification disabled: 

```ts
<ComboBox
    allowFreeText
    maxItems={4}
    onClear={() => {}}
    onOpenChange={() => {}}
    onQueryChange={() => {}}
    onValueChange={() => {}}
    status="initial"
    value=""
  >
    <ComboBox.Icon
      icon={{
        SVG: () => {},
        name: 'SearchIcon'
      }}
      position="start"
    />
  </ComboBox>
```

But with minification enabled, I see this:

```ts
<ComboBox
    allowFreeText
    maxItems={4}
    onClear={function Ga(){}}
    onOpenChange={function Ga(){}}
    onQueryChange={function Ga(){}}
    onValueChange={function Ga(){}}
    status="initial"
    value=""
  >
    <ComboBox.Icon
      icon={{
        SVG: function Ga(){},
        name: 'SearchIcon'
      }}
      position="start"
    />
    // ...

  </ComboBox>
```

ComboBox doesn't have this issue in the Import block, just in the Show Code blocs.
